core/engine/src/bytecompiler/mod.rs

```diff
@@ -1036,65 +1057,71 @@ impl<'ctx> ByteCompiler<'ctx> {
                 let index = self.get_binding(&binding);
                 self.emit_binding_access(BindingAccessOpcode::GetName, &index, dst);
             }
-            Access::Property { access } => match access {
-                PropertyAccess::Simple(access) => {
-                    let object = self.register_allocator.alloc();
-                    self.compile_expr(access.target(), &object);
+            Access::Property { access } => {
+                match access {
+                    PropertyAccess::Simple(access) => {
+                        self.push_source_position(access.field().span().start());
```

Comment:
How complicated would it be to have a `SourcePositionGuard` that on `Drop` would call `pop_source_position`? Seems like it would make the code cleaner (and stronger since you can't forget to pop at the end or on `?`), but it might be impossible with the borrow checker as it is.

Reply:
The way I do this is a `XyzGuard` which takes a mut to a structure, and has `DerefMut` to the structure, then use that guard to access the other methods.